notification-server/src/processor.rs

```diff
@@ -143,31 +156,46 @@ fn map_plt_token_events(
 }
 
 pub async fn process(
-    transactions: impl Stream<Item = Result<types::BlockItemSummary, tonic::Status>>,
-) -> Vec<NotificationInformationBasic> {
-    transactions
-        .filter_map(|result| async move { result.ok() })
-        .flat_map(
-            |t| -> futures::stream::Iter<std::vec::IntoIter<NotificationInformationBasic>> {
-                futures::stream::iter(match &t.details {
-                    types::BlockItemSummaryDetails::AccountTransaction(account_transaction) => {
-                        map_transaction_to_notification_information(
-                            &account_transaction.effects,
-                            t.hash,
+    mut transactions: impl Stream<Item = Result<types::BlockItemSummary, tonic::Status>> + Unpin,
+) -> anyhow::Result<Vec<NotificationInformationBasic>> {
```

Comment:
I don't think this is the way to go, as unknown transaction types can be ignored until we need them IMO.

Reply:
same as above.